### PR TITLE
Skip report execution on curse

### PR DESCRIPTION
**Pull request by RayXpub**
## Motivation

The goal of this PR is to remove reverts on lane cursing for DON execution transactions. If a lane is cursed, reverting will cause the whole execution transaction to fail so any execution reports for other lanes will be blocked.

## Solution

For DON execution transactions the behavior is now to skip the report and emit a `SkippedReportExecution`. For manual execution we still revert, otherwise the transaction will silently fail for the users.

### CODE REVIEW

File: contracts/src/v0.8/ccip/test/offRamp/OffRampSetup.t.sol [hunk 1]
```diff
@@ -501,4 +499,40 @@ contract OffRampSetup is FeeQuoterSetup, MultiOCR3BaseSetup {
       }
     }
   }
+
+  function assertExecutionStateChangedEventLogs(
+    Vm.Log[] memory logs,
```
Comment (RayXpub) [inline contracts/src/v0.8/ccip/test/offRamp/OffRampSetup.t.sol:504]: Required for logs reuse as `vm.getRecordedLogs` consumes the logs so all instances of logs reuse were not performing checks correctly.

File: contracts/src/v0.8/ccip/test/offRamp/OffRamp.t.sol [hunk 2]
```diff
@@ -1230,6 +1262,43 @@ contract OffRamp_batchExecute is OffRampSetup {
...
+    reports[1] = _generateReportFromMessages(SOURCE_CHAIN_SELECTOR_3, messages2);
+
+    vm.recordLogs();
+
+    s_offRamp.batchExecute(reports, new uint256[][](2));
+
+    Vm.Log[] memory logs = vm.getRecordedLogs();
+
+    for (uint256 i = 0; i < logs.length; ++i) {
+      if (logs[i].topics[0] == OffRamp.ExecutionStateChanged.selector) {
```
Comment (elatoskinas) [inline contracts/src/v0.8/ccip/test/offRamp/OffRamp.t.sol:1289]: q: can we re-use the log assertion helper?

File: contracts/src/v0.8/ccip/test/offRamp/OffRamp.t.sol [hunk 3]
```diff
@@ -1230,6 +1262,43 @@ contract OffRamp_batchExecute is OffRampSetup {
...
+    messages1[1] = _generateAny2EVMMessageNoTokens(SOURCE_CHAIN_SELECTOR_1, ON_RAMP_ADDRESS_1, 2);
+    messages2[0] = _generateAny2EVMMessageNoTokens(SOURCE_CHAIN_SELECTOR_3, ON_RAMP_ADDRESS_3, 1);
+
+    Internal.ExecutionReportSingleChain[] memory reports = new Internal.ExecutionReportSingleChain[](2);
+    reports[0] = _generateReportFromMessages(SOURCE_CHAIN_SELECTOR_1, messages1);
+    reports[1] = _generateReportFromMessages(SOURCE_CHAIN_SELECTOR_3, messages2);
+
+    vm.recordLogs();
+
+    s_offRamp.batchExecute(reports, new uint256[][](2));
```
Comment (elatoskinas) [inline contracts/src/v0.8/ccip/test/offRamp/OffRamp.t.sol:1284]: suggestion: should assert emission of the skipped report event

File: contracts/src/v0.8/ccip/offRamp/OffRamp.sol [hunk 4]
```diff
@@ -342,7 +343,17 @@ contract OffRamp is ITypeAndVersion, MultiOCR3Base {
     uint256[] memory manualExecGasLimits
   ) internal {
     uint64 sourceChainSelector = report.sourceChainSelector;
-    _whenNotCursed(sourceChainSelector);
+    bool manualExecution = manualExecGasLimits.length != 0;
+    if (IRMN(i_rmnProxy).isCursed(bytes16(uint128(sourceChainSelector)))) {
+      if (manualExecution) {
```
Comment (elatoskinas) [inline contracts/src/v0.8/ccip/offRamp/OffRamp.sol:348]: nit: `else` could be skipped for more concise formatting:

```solidity
if (manualExecution) {
    revert CursedByRMN(sourceChainSelector);
}

emit SkippedReportExecution(...);
return;

```

File: contracts/src/v0.8/ccip/test/offRamp/OffRamp.t.sol (hunk 2 above)
Comment (RayXpub) [reply contracts/src/v0.8/ccip/test/offRamp/OffRamp.t.sol:1289]: Unfortunately not for this one as here we want to assert that only a specific `ExecutionStateChanged` has been emitted. The helper will not enforce that, the difference is in:
``` solidity
assertEq(logMessageId, messages2[0].header.messageId);
```

The helper uses an if statement:
```solidity
if (logMessageId == messageId)
```

File: contracts/src/v0.8/ccip/offRamp/OffRamp.sol (hunk 4 above)
Comment (RayXpub) [reply contracts/src/v0.8/ccip/offRamp/OffRamp.sol:348]: Updated

File: contracts/src/v0.8/ccip/test/offRamp/OffRamp.t.sol (hunk 3 above)
Comment (RayXpub) [reply contracts/src/v0.8/ccip/test/offRamp/OffRamp.t.sol:1284]: Added

**Review by elatoskinas — APPROVED** [review]

**Review by 0xsuryansh — APPROVED** [review]

**Pull request merged**